### PR TITLE
Admin can Assign the manager to the User, Manager can assign the task…

### PR DESCRIPTION
**Pull request by harmeshverma01**
… to the user

### CODE REVIEW

File: app/views.py [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 from rest_framework import authentication, permissions
-from django.shortcuts import get_object_or_404
 from .utils import admin_required, manager_required
+from rest_framework.authtoken.models import Token
+from rest_framework.response import Response
+from django.contrib.auth import authenticate
+from rest_framework.views import APIView
+from .models import User, Assign, Task
+from rest_framework import status
+from functools import partial
```
Comment (aayush-kumar-codes) [inline app/views.py]: remove unwanted imports

File: app/views.py [hunk 2]
```diff
@@ -57,27 +51,20 @@ def post(self, request):
     #         if serializer.is_valid():
     #             serializer.save()
     #         return Response(serializer.data)
+            
     #     except:
     #         return Response({"message": "Details does not found"}, status=status.HTTP_422_UNPROCESSABLE_ENTITY)
```
Comment (aayush-kumar-codes) [inline app/views.py]: remove these comments

File: app/views.py [hunk 3]
```diff
@@ -88,34 +75,79 @@ def patch(self, request, id=None):
         # return Response({"message": "Details does not found"}, status=status.HTTP_404_NOT_FOUND)
```
Comment (aayush-kumar-codes) [inline app/views.py]: remove comments

File: app/models.py [hunk 4]
```diff
@@ -36,4 +38,16 @@ def has_module_perms(self, app_label):
...
+    
+class Task(models.Model):
+    title = models.CharField(max_length=100)
+    discription = models.CharField(max_length=50, null=True)
+    completed = models.BooleanField(default=False, blank=True)
+    objects = models.Manager()
+    
+    def __str__(self) -> str:
+        return self.title
+
```
Comment (aayush-kumar-codes) [inline app/models.py:48]: create a different app and make Task models and views there.

File: app/views.py (hunk 2 above)
Comment (harmeshverma01) [reply app/views.py]: fixed

File: app/views.py (hunk 3 above)
Comment (harmeshverma01) [reply app/views.py]: fixed

File: app/models.py (hunk 4 above)
Comment (harmeshverma01) [reply app/models.py:48]: fixed

File: Todo_app/urls.py [hunk 5]
```diff
@@ -13,10 +13,12 @@
...
 """
+from django import urls
 from django.contrib import admin
 from django.urls import path, include
 
 urlpatterns = [
     path('admin/', admin.site.urls),
-    path('',include('app.urls') )
+    path('',include('app.urls') ),
+    path('',include('task.urls'))
```
Comment (aayush-kumar-codes) [inline Todo_app/urls.py:23]: there should be any prefix as per functionality like :  /user/

File: app/models.py [hunk 6]
```diff
@@ -15,12 +16,13 @@ class User(AbstractBaseUser):
     id = models.UUIDField(default=uuid.uuid4, primary_key=True)
```
Comment (aayush-kumar-codes) [inline app/models.py:16]: it should be unique true

File: app/static/app/style.css [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    padding:10px 0;
+    border-bottom:1px solid #ddd;
+    text-decoration:line-through;
+}
+#done-items li:last-child{
+    border-bottom:none;
+}
+#checkAll{
+    margin-top:10px;
+}
```
Comment (aayush-kumar-codes) [inline app/static/app/style.css:42]: remove this file, I don't see it's used anywhere.

File: app/template/app/index.html [hunk 8]
```diff
@@ -0,0 +1,49 @@
+{% load static %}
+<html>
+    <head>
+        <link href="//netdna.bootstrapcdn.com/bootstrap/3.2.0/css/bootstrap.min.css" rel="stylesheet" id="bootstrap-css">
+        {% comment %} <link href="{% static 'app/style.css' %}" rel="stylesheet" > {% endcomment %}
+        <script src="//netdna.bootstrapcdn.com/bootstrap/3.2.0/js/bootstrap.min.js"></script>
```
Comment (aayush-kumar-codes) [inline app/template/app/index.html:6]: remove this comment

File: app/template/app/index.html [hunk 9]
```diff
@@ -0,0 +1,49 @@
+{% load static %}
```
Comment (aayush-kumar-codes) [inline app/template/app/index.html:1]: no need to use this tag, since no static file is used.

File: app/urls.py [hunk 10]
```diff
@@ -1,10 +1,13 @@
...
  
 urlpatterns = [
     path('api-users/', Userview.as_view()),
-    path('api/<int:id>/', Userview.as_view()),
     path('login', Loginview.as_view()),
-    path('UserDetails', UserDetailsView.as_view())
+    path('UserDetails/<uuid:id>', UserDetailsView.as_view()),
+    path('Manager', ManagerView.as_view()),
+    path('AssignView', AssignUserView.as_view()),
+    path('', views.getuser, name='getuser')
```
Comment (aayush-kumar-codes) [inline app/urls.py:12]: give some meaningful route to it.

File: task/views.py [hunk 11]
```diff
@@ -0,0 +1,50 @@
+from django.shortcuts import render
+from rest_framework import authentication, permissions
+from task.serializers import TaskSerializer
+from rest_framework.response import Response
+from app.utils import  manager_required
+from .models import Task
+from rest_framework.views import APIView
+from rest_framework import status
+from functools import partial
```
Comment (aayush-kumar-codes) [inline task/views.py:9]: remove this import

Comment (harmeshverma01) [reply task/views.py:9]: Fixed

File: app/views.py (hunk 1 above)
Comment (harmeshverma01) [reply app/views.py]: fixed

File: task/views.py (hunk 11 above)
Comment (harmeshverma01) [reply task/views.py:9]: fixed

File: app/template/app/index.html (hunk 9 above)
Comment (harmeshverma01) [reply app/template/app/index.html:1]: fixed

File: app/template/app/index.html (hunk 8 above)
Comment (harmeshverma01) [reply app/template/app/index.html:6]: fixed

File: app/static/app/style.css (hunk 7 above)
Comment (harmeshverma01) [reply app/static/app/style.css:42]: fixed

File: Todo_app/urls.py (hunk 5 above)
Comment (harmeshverma01) [reply Todo_app/urls.py:23]: fixed

File: app/models.py (hunk 6 above)
Comment (harmeshverma01) [reply app/models.py:16]: fixed

File: app/urls.py (hunk 10 above)
Comment (harmeshverma01) [reply app/urls.py:12]: fixed

**Pull request merged**